echobot/echobot.c

```diff
@@ -40,9 +40,13 @@ Tox *create_tox()
 
         uint8_t *savedata = malloc(fsize);
 
-        fread(savedata, fsize, 1, f);
+        size_t read_size = fread(savedata, fsize, 1, f);
         fclose(f);
 
+        if (read_size != fsize) {
```

Comment:
`fread` returns the number of objects read, so this should be compared with `1`, not `fsize`?

Reply:
Done.